lib/core/class-affiliates-user-registration.php

```diff
@@ -144,9 +144,47 @@ public static function user_register( $user_id, $params = array() ) {
 				)
 			);
 
-			if ( class_exists( 'Affiliates_Referral_WordPress' ) ) {
-				$r = new Affiliates_Referral_WordPress();
-				$affiliate_id = $r->evaluate( $post_id, $description, $data, $base_amount, $amount, $currency, $user_registration_referral_status, self::REFERRAL_TYPE );
+			if ( class_exists( 'Affiliates_Referral_Controller' ) ) {
+				$rc = new Affiliates_Referral_Controller();
+				if ( $rc->evaluate_referrer() ) {
+					$params = $rc->evaluate_referrer();
```

Comment:
Two issues here:
1. Calling the same function twice is redundant if you make an assignment you can just check its value.
2. $params has different semantics that do not correspond to what the method returns.